packages/notifications/push/amplify_push_notifications/test/amplify_push_noitfications_impl_test.dart

```diff
@@ -142,6 +122,26 @@ void main() {
       ).called(1);
     });
 
+    test(
+        'should fire both internal token.first and external token received listener',
```

Comment:
This test actually doesn't verify `token.first` though, can you add assertion for it?